src/Utilities/functions.jl

```diff
@@ -4,20 +4,20 @@ using Compat.Test
 """
     evalvariables(varval::Function, f::AbstractFunction)
 
-Returns the value of function `f` if each variable index `vi` is evaluate as `varval(vi)`.
+Returns the value of function `f` if each variable index `vi` is evaluated as `varval(vi)`.
 """
 function evalvariables end
 evalvariables(varval::Function, f::MOI.SingleVariable) = varval(f.variable)
 evalvariables(varval::Function, f::MOI.VectorOfVariables) = varval.(f.variables)
-evalvariables(varval::Function, f::MOI.ScalarAffineFunction) = sum(evalterm.(varval, f.terms)) + f.constant
+evalvariables(varval::Function, f::MOI.ScalarAffineFunction) = sum(evalterm(varval, t) for t in f.terms) + f.constant
```

Comment:
Why is this preferred?

Reply:
It doesn't allocate, it just sums the terms.

Reply:
Actually it seems like it does, I will double check.

Reply:
Nah was just bechmarking in global scope, the new one doesn't allocate:
```julia
julia> using BenchmarkTools

julia> a = rand(1000);

julia> f(a) = sum(i^2 for i in a)
f (generic function with 1 method)

julia> g(a) = sum(a.^2)
g (generic function with 1 method)

julia> @btime f(a);
  1.422 μs (3 allocations: 48 bytes)

julia> @btime g(a);
  6.199 μs (26 allocations: 8.98 KiB)
```

Reply:
Even better is `mapreduce`: 
```julia
julia> @btime mapreduce((x)->(x^2), +, $a)
  89.555 ns (0 allocations: 0 bytes)
329.62559257013555

julia> @btime f($a);
  1.130 μs (3 allocations: 48 bytes)

julia> @btime g($a);
  600.012 ns (1 allocation: 7.94 KiB)
```